src/main/webapp/app/exercises/shared/exercise-categories/exercise-categories.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, Input, OnChanges, OnDestroy, OnInit } from '@angular/core';
+import { DifficultyLevel, Exercise, IncludedInOverallScore } from 'app/entities/exercise.model';
+import dayjs from 'dayjs/esm';
+import { QuizExercise } from 'app/entities/quiz/quiz-exercise.model';
+import { Subscription } from 'rxjs';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+    selector: 'jhi-exercise-categories',
+    templateUrl: './exercise-categories.component.html',
+    styleUrls: ['./exercise-categories.component.scss'],
+})
+export class ExerciseCategoriesComponent implements OnInit, OnDestroy, OnChanges {
+    readonly IncludedInOverallScore = IncludedInOverallScore;
+    readonly dayjs = dayjs;
+
+    @Input() exercise: Exercise;
+
+    @Input() shouldShowNotReleasedTag = false;
+    @Input() shouldShowQuizLiveTag = false;
+    @Input() shouldShowDifficultyTag = false;
+    @Input() shouldShowDifficultyTagIfNoLevel = false;
+    @Input() shouldShowIncludedInScoreBadge = false;
+
+    asQuizExercise(exercise: Exercise): QuizExercise {
+        return exercise as QuizExercise;
+    }
+
+    translatedDifficulty: string;
```

Comment:
Why are you using these helper attributes? especially for something like `not released` you should also be able to directly pipe the translation key into `artemisTranslate` in the template.

Reply:
Thank you! You are right, the translatedNotReleased helper attribute as well as the tooltips should directly be piped into artemisTranslate and I have to fix that. :)

What do you think about translatedIncludedAsBonus and translatedDifficulty. Their translation depends on their corresponding enums. Should they be translated differently or is it ok if they are translated in the ExerciseCategoriesComponent?